x-pack/plugins/apm/public/components/app/ServiceDetails/ServiceIntegrations/MachineLearningFlyout.tsx

```diff
@@ -0,0 +1,256 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  EuiButton,
+  EuiCallOut,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFlyout,
+  EuiFlyoutBody,
+  EuiFlyoutFooter,
+  EuiFlyoutHeader,
+  EuiSpacer,
+  EuiText,
+  EuiTitle
+} from '@elastic/eui';
+import React, { Component } from 'react';
+import { toastNotifications } from 'ui/notify';
+// @ts-ignore
+import {
+  getMlJob,
+  startMlJob
+} from 'x-pack/plugins/apm/public/services/rest/ml';
+// @ts-ignore
+import { getAPMIndexPattern } from 'x-pack/plugins/apm/public/services/rest/savedObjects';
+import { KibanaLink, ViewMLJob } from 'x-pack/plugins/apm/public/utils/url';
+
+interface FlyoutProps {
+  isOpen: boolean;
+  onClose: () => void;
+  serviceName: string;
+  transactionType?: string;
+  location: any;
+}
+
+interface FlyoutState {
+  isLoading: boolean;
+  hasMlJob: boolean;
+  hasIndexPattern: boolean | null;
```

Comment:
This doesn't actually need to be `null`. It looks like the render method only checks that `hasIndexPattern` is falsy, so it could be initialized as `false` and could avoid the awkward tertiary nature of this union type.